src/main/scala/net/fehmicansaglam/akkahttpclient/SimpleHttpRequestBuilder.scala

```diff
@@ -33,6 +33,14 @@ case class SimpleHttpRequestBuilder(request: HttpRequest) {
     SimpleHttpRequestBuilder(request.withUri(request.uri.withQuery(query)))
   }
 
+  def headers(kvs: (String, String)*): SimpleHttpRequestBuilder = {
+    val customHeaders = kvs.map {
+      case (key, value) => RawHeader(key, value).asInstanceOf[HttpHeader]
+    }
+
+    SimpleHttpRequestBuilder(request.withHeaders(customHeaders.to[collection.immutable.Seq]))
```

Comment:
@ertugrulyilmaz  Your code overwrites current headers. Below is a better alternative:

```scala
  def headers(kvs: (String, String)*): SimpleHttpRequestBuilder = {
    SimpleHttpRequestBuilder(request.mapHeaders(_ ++ kvs.map((RawHeader.apply _).tupled)))
  }
```